co-authors-plus.php

```diff
@@ -1474,13 +1475,62 @@ public function filter_jetpack_open_graph_tags( $og_tags, $image_dimensions ) {
 	}
 
 	/**
+	 * Retrieve a list of coauthor terms for a single post.
+	 *
+	 * Grabs a correctly ordered list of authors for a single post, appropriately
+	 * cached because it requires `wp_get_object_terms()` to succeed.
+	 *
+	 * @param int $post_id ID of the post for which to retrieve authors.
+	 * @return array Array of coauthor WP_Term objects
+	 */
+	public function get_coauthor_terms_for_post( $post_id ) {
+
+		if ( ! $post_id ) {
+			return array();
+		}
+
+		$cache_key = 'coauthors_post_' . $post_id;
+		$coauthor_terms = wp_cache_get( $cache_key, 'co-authors-plus' );
+
+		if ( false === $coauthor_terms ) {
+			$cached = wp_get_object_terms( $post_id, $this->coauthor_taxonomy, array(
+				'orderby' => 'term_order',
+				'order' => 'ASC',
+			) );
+			// Cache an empty array if the taxonomy doesn't exist.
+			$coauthor_terms = ( is_wp_error( $cached ) ) ? array() : $cached;
+			wp_cache_set( $cache_key, $coauthor_terms, 'co-authors-plus' );
```

Comment:
My worry about not caching errors is that a simple typo could easily result in `wp_get_object_terms()` triggering a lot more than we'd like, and the uncached queries that comes with it.